README.md

```diff
@@ -21,135 +21,171 @@ gem 'acme-client'
 
-# You are not required to send a Content-Type. This method will return the right Content-Type should you decide to include one.
-challenge.content_type
+Once your are ready to complete the challenge, you can request to server to perform the verification.
 
-# Save the file. We'll create a public directory to serve it from, and inside it we'll create the challenge file.
-FileUtils.mkdir_p( File.join( 'public', File.dirname( challenge.filename ) ) )
+```ruby
+challenge.request_validation
+```
 
-# We'll write the content of the file
-File.write( File.join( 'public', challenge.filename), challenge.file_content )
+The validation is performed asynchronously and can take some time to be performed by the server.
 
-# Optionally save the authorization URI for use at another time (eg: by a background job processor)
-File.write('authorization_uri', authorization.uri)
+You can poll until its status change.
 
-# The challenge file can be served with a Ruby webserver.
-# You can run a webserver in another console for that purpose. You may need to forward ports on your router.
-#
-# $ ruby -run -e httpd public -p 8080 --bind-address 0.0.0.0
+```ruby
+until challenge.status != 'pending'
+  sleep(2)
+  challenge.reload
+end
+challenge.status # => 'valid'
+```
 
-# Load a challenge based on stored authorization URI. This is only required if you need to reuse a challenge as outlined above.
-challenge = client.fetch_authorization(File.read('authorization_uri')).http01
+### Downloading a certificate
 
-# Once you are ready to serve the confirmation request you can proceed.
-challenge.request_verification # => true
-challenge.authorization.verify_status # => 'pending'
+Once all required authorizations have been validated through challenges, the order can be finalized using a CSR ([Certificate Signing Request](https://en.wikipedia.org/wiki/Certificate_signing_request)).
 
-# Wait a bit for the server to make the request, or just blink. It should be fast.
-sleep(1)
+CSR can be slightly tricky to generate using OpenSSL from Ruby standard library. The `acme-client` provide a utility class `CertificateRequest` to help with that.
 
-# Rely on authorization.verify_status more than on challenge.verify_status,
-# if the former is 'valid' you can already issue a certificate and the status of
-# the challenge is not relevant and in fact may never change from pending.
-challenge.authorization.verify_status # => 'valid'
-challenge.error # => nil
+The certificate generation is happening asynchronously. You may need to poll.
 
-# If authorization.verify_status is 'invalid', you can get at the error
-# message only through the failed challenge.
-authorization.verify_status # => 'invalid'
-authorization.http01.error # => {"type" => "...", "detail" => "..."}
+```ruby
+csr = Acme::Client::CertificateRequest.new(private_key: private_key, subject: { common_name: 'example.com' })
+order.finalize(csr: csr)
+until order.status != 'processing'
+  sleep(1)
+end
```

Comment:
Style preference, but I think you could do this as a one-liner:

```ruby
sleep(1) while order.status == 'processing'
```

Also, `while` might be clearer than `until`, given that there is essentially a double negative between `until` and `!=`.